ngraph/python/setup.py

```diff
@@ -237,130 +49,128 @@ def cpp_flag(compiler):
-        self._add_extra_compile_arg("/GL", ext.extra_compile_args)  # Whole Program Optimization
-        self._add_extra_compile_arg("/analyze", ext.extra_compile_args)
-
-    def _add_unix_compiler_flags(self, ext):
-        if not self._add_extra_compile_arg("-fstack-protector-strong", ext.extra_compile_args):
-            self._add_extra_compile_arg("-fstack-protector", ext.extra_compile_args)
-
-        self._add_extra_compile_arg("-fvisibility=hidden", ext.extra_compile_args)
-        self._add_extra_compile_arg("-flto", ext.extra_compile_args)
-        self._add_extra_compile_arg("-fPIC", ext.extra_compile_args)
-
-        ext.extra_compile_args += ["-Wformat", "-Wformat-security"]
-
-    def _customize_compiler_flags(self):
-        """Modify standard compiler flags."""
-        try:
-            # -Wstrict-prototypes is not a valid option for c++
-            self.compiler.compiler_so.remove("-Wstrict-prototypes")
+            build_types = [item.lower() for item in self.cmake_build_types]
+            try:
+                i = build_types.index(str(self.config).lower())
+                self.config = self.cmake_build_types[i]
+            except ValueError:
+                self.announce("Unsupported CMAKE_BUILD_TYPE value: " + self.config, level=4)
+                self.announce("Supported values: {}".format(", ".join(self.cmake_build_types)), level=4)
+                sys.exit(1)
+        super().finalize_options()
+        
+    def run(self):
+        for extension in self.extensions:
+            if extension.name == "_pyngraph":
+                self.build_cmake(extension)
+
+    def build_cmake(self, extension: Extension):
+        """
+        Steps required to build the extension
+        """
+
+        self.announce("Preparing the build environment", level=3)
+
+        build_dir = pathlib.Path(self.build_temp)
+
+        extension_path = pathlib.Path(self.get_ext_fullpath(extension.name))
+
+        os.makedirs(build_dir, exist_ok=True)
+        os.makedirs(extension_path.parent.absolute(), exist_ok=True)
+        
+        # If ngraph_DIR is not set try to build from OpenVINO root
+        root_dir = OPENVINO_ROOT_DIR
+        bin_dir = os.path.join(OPENVINO_ROOT_DIR, "bin")
+        if os.environ.get('ngraph_DIR') is not None:
+            root_dir = PYNGRAPH_ROOT_DIR
+            bin_dir = build_dir
+            
+        self.announce("Configuring cmake project", level=3)
+
+        self.spawn(["cmake", "-H" + root_dir, "-B" + self.build_temp,
+                    "-DCMAKE_BUILD_TYPE={}".format(self.config),
+                    "-DENABLE_CLDNN=OFF",
+                    "-DENABLE_OPENCV=OFF",
+                    "-DENABLE_VPU=OFF",
+                    "-DNGRAPH_PYTHON_BUILD_ENABLE=ON",
+                    "-DNGRAPH_ONNX_IMPORT_ENABLE=ON"])
+        
+        self.announce("Building binaries", level=3)
+        
+        self.spawn(["cmake", "--build", self.build_temp, "--target", extension.name,
```

Comment:
please add parallel build -j $(nproc) - currently build time takes very long time. If there is a way to provide the flag from user perspective - please describe this

Reply:
done. please approve